Sound effects

timotiousPrime opened the pull request:
This is still a WIP. I've uploaded the files for the different sound effects and created the basic functions to play them. I still need to find out where and how I'm going to add them in.

timotiousPrime committed:
added ids for soundEffects

timotiousPrime committed:
functions to play sound effects

timotiousPrime committed:
added sounds to dom

timotiousPrime committed:
condition to play game over sound

timotiousPrime committed:
audio for sound effects

timotiousPrime committed:
game over audio

timotiousPrime committed:
removed console.log score

timotiousPrime committed:
explosion sound

timotiousPrime committed:
add sound functions to rocket event functions

timotiousPrime committed:
add basket moving sound to arrow key press EL

timotiousPrime committed:
made sure all the correct sounds are in the DOM

timotiousPrime committed:
add renderSound func for sound effects if unmuted

timotiousPrime committed:
removed console log of FO

timotiousPrime committed:
catching sound effect

timotiousPrime committed:
falling object spawn sound effect

poltak commented on dom.js:
```diff
@@ -69,4 +75,61 @@ export function playBackgroundMusic(isMuted) {
+    levelUpSound.play()
+}
+
+export function SpawnFallingObjectSound(isMuted) {
```
Just a minor naming issue: this function is named inconsistently compared to the rest.

Also a general naming note: it's generally good practice to name functions starting with a verb. Because they almost always exist to perform some action.So in these cases I would have called these `playLevelUpSound`, etc.

poltak commented on dom.js:
```diff
@@ -69,4 +75,61 @@ export function playBackgroundMusic(isMuted) {
+
+export function gameOverSound(isMuted) {
+    const gameOverMusic = document.getElementById(EL_IDS.gameOverMusic)
+    if (!gameOverMusic) return
```
All these functions are structured the exact same way. So you could "DRY-out" a lot of this code by creating a single function that accepts both `isMuted` and `elementId` then each of these functions can just call that, passing in the specific `EL_IDS` they need to operate on.

Again, not super important

timotiousPrime replied:
Thanks for pointing that out. I went and did what you said. and added most of the functions to the state mutators with one function

timotiousPrime committed:
removed repeated code, simplified sound effs func

timotiousPrime committed:
removed sound effects from event listeners

timotiousPrime committed:
updated sound effects function when FO is caught

timotiousPrime committed:
remov as much sound effects from this file as poss

timotiousPrime committed:
added sound effects to state mutators

timotiousPrime commented:
So I only have one more sound effect to add in, and still unsure how to do it. It's the level up sound effect. The structure of the state mutators doesn't update once per level, instead checks the score to the level and then updates the level every time an object is caught. Maybe one of you can give me a tip on how to solve this and I will. Also I'm not sure why when I try to mute the background music in the state mutator when its game over, so that only the game over music plays, it never works. I had to keep in my renderSounds function for when this happens. Any tips here as well would also be appreciated

poltak committed:
Add in triggering of level up sound effect

poltak committed:
Unify playing of game over music with playing of BG music

- Both of these are long playing tracks, as opposed to short sound effects
- They can also either have one of the two playing, or none if muted, but never both at the same time
- Hence managing them from the same place seems to make most sense, to contain all the conditional logic in one place
- I also properly set them up to reset their tracks before they begin playing, so you don't start a new game and have the BG music start playing from where it left off and vice-versa with game over

poltak committed:
Move catch obj sound triggering from falling logic to mutator

- now they're all in the mutators

poltak committed:
Rename playSoundEffect's element param to elementId

- an element is what would be returned from `getElementById`, while it would expect an element's ID

timotiousPrime commented:
Thanks for your commits Jon! Works great now!

poltak committed:
Remove unused params

poltak reviewed (approved):
Just added one minor change, removing some unused function params that got added.

Apart from that, I'm happy to merge this!
